cmd/blocktest/main.go

```diff
@@ -103,103 +96,44 @@ type Test struct {
 	Pre                map[string]Account
 }
 
-var (
-	Identifier       string
-	KeyRing          string
-	DiffTool         bool
-	DiffType         string
-	KeyStore         string
-	StartRpc         bool
-	StartWebSockets  bool
-	RpcListenAddress string
-	RpcPort          int
-	WsPort           int
-	OutboundPort     string
-	ShowGenesis      bool
-	AddPeer          string
-	MaxPeer          int
-	GenAddr          bool
-	BootNodes        string
-	NodeKey          *ecdsa.PrivateKey
-	NAT              nat.Interface
-	SecretFile       string
-	ExportDir        string
-	NonInteractive   bool
-	Datadir          string
-	LogFile          string
-	ConfigFile       string
-	DebugFile        string
-	LogLevel         int
-	LogFormat        string
-	Dump             bool
-	DumpHash         string
-	DumpNumber       int
-	VmType           int
-	ImportChain      string
-	SHH              bool
-	Dial             bool
-	PrintVersion     bool
-	MinerThreads     int
-)
-
-// flags specific to cli client
-var (
-	StartMining    bool
-	StartJsConsole bool
-	InputFile      string
-)
-
 func main() {
-	init_vars()
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "%s <testfile>\n", os.Args[0])
+		flag.PrintDefaults()
+	}
+	flag.Parse()
 
-	Init()
+	runtime.GOMAXPROCS(runtime.NumCPU())
+	logger.AddLogSystem(logger.NewStdLogSystem(os.Stderr, log.LstdFlags, logger.DebugDetailLevel))
+	defer func() { logger.Flush() }()
 
-	if len(TestFile) < 1 {
-		log.Fatal("Please specify test file")
+	if len(os.Args) < 2 {
+		utils.Fatalf("Please specify a test file as the first argument.")
 	}
-	blocks, err := loadBlocksFromTestFile(TestFile)
+	blocks, err := loadBlocksFromTestFile(os.Args[1])
 	if err != nil {
-		panic(err)
+		utils.Fatalf("Could not load blocks: %v", err)
 	}
 
-	runtime.GOMAXPROCS(runtime.NumCPU())
+	chain := memchain()
```

Comment:
We also need RPC API started, as we want the test to end with enough running to allow ethereum.js to call RPC API to test stuff.